### PR TITLE
V0.4.3

### PR DESCRIPTION
**Pull request by ccuilla**
- Updated packages and dependencies
- Updated Airbnb eslint rules but disabled a couple and fixed issues for one rule
- Add `yarn.lock`

### CODE REVIEW

File: src/RegularExpression.js [hunk 1]
```diff
@@ -2,8 +2,11 @@ import { GraphQLScalarType } from 'graphql';
 import { GraphQLError } from 'graphql/error';
 import { Kind } from 'graphql/language';
 
-function RegularExpression(name, regex) {
+function RegularExpression(name, regex, options = {}) {
   const REGEX = new RegExp(regex);
+  const errorMessage = options.errorMessage
```
Comment (cfnelson) [inline src/RegularExpression.js]: You may want to guard against `null` being passed here to stop an error from throwing when `options.errorMessage` is executed.

If you want to go a step further, but maybe overkill you could do a type check to ensure its an object. May be overkill 🤷‍♂️

File: src/__tests__/DateTime.test.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 import { Kind } from 'graphql/language';
 
-import { DateTime } from '../';
+import { DateTime } from '..';
```
Comment (cfnelson) [inline src/__tests__/DateTime.test.js]: I had no idea that this was possible. 🤦‍♂️  😂 
Learnt something new today. 🥇

File: .eslintrc.js [hunk 3]
```diff
@@ -1,7 +1,8 @@
 module.exports = {
   extends: 'airbnb',
   rules: {
-    "linebreak-style": 'off', // Airbnb specifies Unix style line endings, turn this off for Windows users
+    'implicit-arrow-linebreak': 'off',
+    'linebreak-style': 'off', // Airbnb specifies Unix style line endings, turn this off for Windows users
```
Comment (cfnelson) [inline .eslintrc.js]: Not related to this PR but I am curious as to why this is switched off, wouldn't this allow different types of line endings to be committed into code if someone using a windows os contributes to the pkg?

File: src/__tests__/RegularExpression.test.js [hunk 4]
```diff
@@ -44,6 +42,56 @@ describe('RegularExpression', () => {
       });
     });
 
+    describe('does not match (w/options)', () => {
```
Comment (cfnelson) [inline src/__tests__/RegularExpression.test.js]: Maybe add a test when null/undefined or even another data type is passed if you want to be that verbose? 🤷‍♂️

File: package.json [hunk 5]
```diff
@@ -28,19 +28,19 @@
...
     "eslint-plugin-react": "^7.7.0",
-    "graphql": "^0.13.1",
-    "jest": "^22.4.2",
+    "graphql": ">=14.1.0",
+    "jest": "^23.6.0",
     "nodemon": "^1.17.2"
   },
   "peerDependencies": {
-    "graphql": "^0.13.1"
+    "graphql": ">=14.1.0"
```
Comment (cfnelson) [inline package.json]: Do you not want to still support the older graphql `0.13.x` version any longer?

File: src/RegularExpression.js (hunk 1 above)
Comment (ccuilla) [reply src/RegularExpression.js]:
```
  const errorMessage = options.errorMessage
    ? options.errorMessage
    : (r, v) => `Value does not match the regular expression ${r}: ${v}`;
```

Though that could be simplified to:

```
  const errorMessage = options.errorMessage || (r, v) => `Value does not match the regular expression ${r}: ${v}`;
```

File: src/__tests__/DateTime.test.js (hunk 2 above)
Comment (ccuilla) [reply src/__tests__/DateTime.test.js]: Me too.

File: .eslintrc.js (hunk 3 above)
Comment (ccuilla) [reply .eslintrc.js]: Because it would have forced me to go through all my code and reformat it. This also might be in conflict with Prettier.

File: package.json (hunk 5 above)
Comment (ccuilla) [reply package.json]: Hmmm. Good question.

Comment (capaj) [reply package.json]: I wouldn't bother. All the people who run GQL servers usually upgrade fairly regularly, so I don't know anyone who is still using `0.13.x.`

**Comment by capaj** [comment]
@cfnelson yeah you're certainly not supposed to have both in your repo. It's a massive pain to keep both in sync. Just pick one and stick to that.

File: package.json (hunk 5 above)
Comment (ccuilla) [reply package.json]: We went with `>=0.13.1` for now since I don't think it hurts anything. It just means that this package will work with `0.13.1` or greater. At some point, if we add features that _require_ a later version, then we'll bump this.
